Realm/Realm/Logging/Logger.cs

```diff
@@ -120,12 +157,19 @@ protected Logger()
 
         internal static void LogDefault(LogLevel level, string message) => Default?.Log(level, message);
 
+        internal static void LogDefault(LogLevel level, LogCategory category, string message) => Default?.Log(level, category, message);
+
         /// <summary>
         /// Log a message at the supplied level.
         /// </summary>
         /// <param name="level">The criticality level for the message.</param>
         /// <param name="message">The message to log.</param>
         public void Log(LogLevel level, string message)
```

Comment:
This should be deprecated.

Reply:
We could deprecate it, but then they'd always need to provide the category if using the `public void Log(LogLevel level, LogCategory category, string message)` signature. Kotlin and Swift uses `category, level, message`, so it's somewhat inconsistent with the `SetLogLevel()` args, I believe Dart uses `level, message, optional category` (so also inconsistent but provides optionality).

Reply:
We can take the dart approach and add an optional category to the log method.

Reply:
🤟

Reply:
It's been [updated](https://github.com/realm/realm-dotnet/pull/3634/commits/ab3ee611393e570d3ac020d417e68f4f89c27330) to use the category as an optional 3rd param. I think it feels slightly off logging with the message in between the level and category, but could be okay.